src/DelegateDecompiler/Processor.cs

```diff
@@ -79,12 +79,20 @@ public void Merge(Expression test, ProcessorState leftState, ProcessorState righ
 
             public Expression Final()
             {
-                return Stack.Count == 0
-                       ? Expression.Empty()
-                       : Stack.Pop();
+                if (Stack.Count == 0) return Expression.Empty();
+                if (Stack.Count == 1) return Stack.Pop();
+
+                // Resolves hazzik#122 
+                //TODO confirm what state the stack should be at the end
+                var expressions = Stack.Reverse()
+                    .Select(it => it.Expression)
+                    .Where(it => it.NodeType != ExpressionType.NewArrayBounds);
```

Comment:
why?

Reply:
Ok, I know why. Because `RuntimeMethods.InitializeArray` handling is lying and returns array, when it, in fact, return void.

Reply:
Anyway. This code is wrong, and the issue should be fixed at source.